add section detailing CCI config for Fastlane deployment examples (#7525)
circleci/circleci-docs: jekyll/_cci2/deploy-ios-applications.md

{: #overview }
{:.no_toc}

Using Fastlane, CircleCI can automatically deploy iOS apps to various services. This helps remove the manual steps required to ship a beta or release version of an iOS app to its intended audience.

Deployment _lanes_ can be combined with testing _lanes_ so that the app is automatically deployed upon a successful build and test.

> Using the deployment examples on this page requires that code signing is already configured for your project. To learn how to set up code signing, see the [Setting Up Code Signing]({{site.baseurl}}/ios-codesigning/) page.

Reviewer: ```suggestion
Using the deployment examples on this page requires that code signing is already configured for your project. To learn how to set up code signing, see the [Setting up code signing]({{site.baseurl}}/ios-codesigning/) page.
```

)
```

## CircleCI config for Fastlane integration
{: #circleci-config-for-fastlane-integration }

> All the examples on this page use Fastlane to configure deployment. For each example the following example `.circleci/config.yml` configuration can be used to integrate your Fastlane setup with CircleCI. This is an example config which should be edited to fot the needs of your project:

Reviewer: ```suggestion
All the examples on this page use Fastlane to configure deployment. For each example the following example `.circleci/config.yml` configuration can be used to integrate your Fastlane setup with CircleCI. This is an example config which should be edited to the needs of your project:
```

> The environment variable `FL_OUTPUT_DIR` is the artifact directory where FastLane logs and a signed `.ipa` file should be stored. Use this to set the path in the `store_artifacts` step to automatically save logs and build artifacts from Fastlane.

Reviewer: ```suggestion
The environment variable `FL_OUTPUT_DIR` is the artifact directory where FastLane logs, and a signed `.ipa` file should be stored. Use this to set the path in the `store_artifacts` step to automatically save logs and build artifacts from Fastlane.
```

{: class="alert alert-note"}

```yaml
# .circleci/config.yml
version: 2.1
jobs:
  build-and-test:
    macos:
      xcode: 12.5.1
    environment:
      FL_OUTPUT_DIR: output
      FASTLANE_LANE: test
    steps:
      - checkout
      - run: bundle install
      - run:
          name: Fastlane
          command: bundle exec fastlane $FASTLANE_LANE
      - store_artifacts:
          path: output
      - store_test_results:
          path: output/scan

  adhoc:
    macos:
      xcode: 12.5.1
    environment:
      FL_OUTPUT_DIR: output
      FASTLANE_LANE: adhoc
    steps:
      - checkout
      - run: bundle install
      - run:
          name: Fastlane
          command: bundle exec fastlane $FASTLANE_LANE
      - store_artifacts:
          path: output

workflows:
  build-test-adhoc:
    jobs:
      - build-and-test
      - adhoc:
          filters:
            branches:
              only: development
          requires:
            - build-and-test
```

> In this example, the `adhoc` job exists to enable producing a development build, or upload to Testflight, on pushes to a dev branch, in this example a branch called `development`.

Reviewer: This sentence is a little awkward in the flow.
> In this example, the `adhoc` job exists to enable producing a development build, or upload to Testflight, on pushes to a dev branch, in this example a branch called `development`.

Maybe?

```suggestion
In this example, upon pushing to a development branch, the `adhoc` job enables producing a development build, or upload to Testflight.
```
